### ISSUE TITLE
Check `xl_sheets_to_csv()` with trial data collected at SCBI

### ISSUE THREAD

**Issue by maurolepore** [forestgeo/FastField#7]

**Comment by maurolepore** [comment]
@jess-sue,

The code fails

Wth the data from SCBI that you sent me today, the code fails because the spreadsheet `start_page` is missing. `start_page`. Any idea why is this that case?

```R
library(FastField)
# This should be the path to the directory containing the excel files
sheets_directory <- here::here("inst/7_scbi/input")
# This should be the path to the directory to output the .csv files
output_directory <- here::here("inst/7_scbi/output")
# Do the work
xl_sheets_to_csv(sheets_directory, output_directory)

#> Error: Data should contain these sheets:
#>   start_page, orignal_stems, new_secondary_stems, recruits
#> * Sheets found: root, existing_stems, field_meta_data, form_meta_data, 
#> new_secondary_stems, orignal_stems, recruits 
```

**Comment by jess-shue** [comment]
It looks like the sheet names have changed due to a change in the form - start page no longer exists. I will look into this more to provide updated sheet names. The "?" will be removed from the column header "New_Stem?" and won't be an issue going forward.

**Comment by jess-shue** [comment]
It appears that FFF has updated how the data is exported. Instead of 'start_page' being the necessary worksheet to find the quadrat, date, and team it is now all exported into the 'Root' worksheet.

**Comment by maurolepore** [comment]
Great, I'll adapt the code accordingly. Thanks.

**Comment by maurolepore** [comment]
@jess-shue, 

Unfortunately the data from SCBI that you sent me isn't suitable for this issue. The column `New Stem` is problematic -- not only because of the ? but also because all values are `0`. I'll write an informative error message for when this happens. But could you please send me a new dataset? The closest it is to a real one the better I'll be able to help you by developing a function that works as expected.

![](https://i.imgur.com/4sRwjBH.png)

**Comment by maurolepore** [comment]
Thanks @jess-shue for your quick reply.

And thanks for your patience in explaining this again. I understand the meaning of this column. Sorry I wasnt' clear enough. What I meant is that the code brakes if ALL values of `New Stem` are `0.`  At least 1 value must be non-cero. Now the code brakes more elegantly because I added an informative error message. But still fails. 

I can change some `0`s to `1`s myself -- like this:

![](https://i.imgur.com/Wk0Nyz1.png)

The problem with this is that the data isn't real data. Using an artificial dataset -- that I myself manipulate -- is a much less good option compared to using a real dataset. I know it's annoying but if you can create the dataset with the app and then send it to me, we will be able to trust the code much more than if I just create what _I think_ the dataset will look like. 

An example of how problematic this might be is exactly what you commented above: You discovered that FastField decided to remove on sheet from the output dataset. We would have never imagined that.

So in short, can you please use the app to create at least 3 lines of observations and send me the excel dataset?

**Comment by maurolepore** [comment]
THANKS @jess-shue, I ANSWER IN CAPS BELOW.

```R
From: Shue, Jessica <ShueJ@si.edu>
Date: Tue, Mar 20, 2018 at 11:55 AM
Subject: new dataset
```
> ... Attached is an updated example with two “secondaries” (I just made them up).

THANKS!

> In order to fluidly add new secondaries in the app I had to format it this way – which created the fake records. The New Stem column was added to quickly tell us when a stem was added (1) vs. when it is a fake record (0). 

I DON'T FULLY UNDERSTAND THIS BUT I THINK IT DOESN'T MATTER -- IT SEEMS ITS JUST ABOUT HOW YOU TRICKED THE SYSTEM TO GET WHAT YOU WANTED.

> There will likely be times when the New Stem value will be ‘0’ for every record in that submission like the data from March 15th. All of those records should be removed.

THANKS! YOU ARE TOTALLY RIGHT. THEN I NEED TO TWEAK THE CODE TO ACCOUNT FOR THIS CASE.

If this isn’t possible I will try to find another solution. I had asked FFF about this problem before with no result, but I can ask again!

LET ME TRY BEFORE YOU BOTHER. I'LL KEEP YOU POSTED.

**Comment by jess-shue** [comment]
Thanks! I can explain the formatting in the app further if that would help...let me know!

**Comment by maurolepore** [comment]
Now the code (passed but it) throws a warning if it detects no new stem. 
```R
file_path <- here::here("inst/7_scbi/input_0/612_03-15-2018_MS_EG.xlsx")
new_secondary_stems <- read_excel(file_path, sheet = "new_secondary_stems")
unique(new_secondary_stems$`New Stem`)
## [1] "0"

xl_sheets_to_csv(here::here("inst/7_scbi/input_0/"))
## Warning: Deceted no new stem in `new_secondary_stems`.
```
